persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/field/PatternField.scala

```diff
@@ -48,27 +40,42 @@ class PatternField[OwnerType <: BsonRecord[OwnerType]](rec: OwnerType)
     case o => setFromString(o.toString)
   }
 
-  def setFromJValue(jvalue: JValue): Box[Pattern] = jvalue match {
+  override def setFromJValue(jvalue: JValue): Box[Pattern] = jvalue match {
     case JNothing|JNull if optional_? => setBox(Empty)
     case JObject(JField("$regex", JString(s)) :: JField("$flags", JInt(f)) :: Nil) =>
       setBox(Full(Pattern.compile(s, f.intValue)))
     case other => setBox(FieldHelpers.expectedA("JObject", other))
   }
 
   // parse String into a JObject
-  def setFromString(in: String): Box[Pattern] = tryo(JsonParser.parse(in)) match {
+  override def setFromString(in: String): Box[Pattern] = tryo(JsonParser.parse(in)) match {
     case Full(jv: JValue) => setFromJValue(jv)
     case f: Failure => setBox(f)
     case other => setBox(Failure("Error parsing String into a JValue: "+in))
   }
 
-  def toForm: Box[NodeSeq] = Empty
+  override def toForm: Box[NodeSeq] = Empty
 
-  def asJs = asJValue match {
+  override def asJs = asJValue match {
     case JNothing => JsNull
     case jv => Str(compactRender(jv))
   }
 
   def asJValue: JValue = valueBox.map(v => JsonRegex(v)) openOr (JNothing: JValue)
 }
 
+class PatternField[OwnerType <: BsonRecord[OwnerType]](owner: OwnerType) extends PatternTypedField[OwnerType](owner) with MandatoryTypedField[Pattern] {
```

Comment:
Yep. Will change this. Would you say using the `@deprecatedName` annotation is a good idea? I think this recognizing the parameter as the owner right here in the constructor is a bit better.